pkg/git/v1/transport.go

```diff
@@ -47,6 +48,10 @@ func AuthSecretStrategyForURL(URL string) (common.AuthSecretStrategy, error) {
 type BasicAuth struct{}
 
 func (s *BasicAuth) Method(secret corev1.Secret) (*common.Auth, error) {
+	if _, ok := secret.Data[common.CACert]; ok {
+		return nil, errors.New("git2go HTTP transport does not support custom certificates.")
```

Comment:
This is go-git, we should merge https://github.com/fluxcd/source-controller/pull/270 to avoid confusion in the future.

Reply:
I propose this error message:
```
found caCert key in secret <name> but go-git HTTP transport does not support custom certificates.
```